core/include/drivers/imx_i2c.h

```diff
@@ -0,0 +1,17 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2020 Foundries.io
+ * Jorge Ramirez-Ortiz <jorge@foundries.io>
+ */
+#ifndef __DRIVERS_IMX_I2C_H
+#define __DRIVERS_IMX_I2C_H
+
+#include <stdint.h>
+#include <tee_api_types.h>
+
+TEE_Result imx_i2c_write(uint8_t bid, uint8_t chip, const uint8_t *p, int l);
+TEE_Result imx_i2c_read(uint8_t bid, uint8_t chip, uint8_t *p, int l);
+TEE_Result imx_i2c_probe(uint8_t bid, uint8_t chip);
```

Comment:
sure but why does it matter? we are never going to have more than 255 buses or 255 chips on a bus...I mean, why int in particular and not some other type? what do we gain?

Reply:
It's likely that requesting here a 1byte data instead of the CPU native integer (unsigned or not) will request few more assembly instructions, without much benefit.
That's said, feel free to keep implementation as is.